src/main/java/com/fasterxml/jackson/core/json/UTF8StreamJsonParser.java

```diff
@@ -2188,10 +2195,11 @@ protected String _handleOddName(int ch) throws IOException
 
         if (currQuadBytes > 0) {
             if (qlen >= quads.length) {
-                _quadBuffer = quads = growArrayBy(quads, quads.length);
+                _quadBuffer = quads = growArrayWithNameLenCheck(quads, quads.length);
             }
             quads[qlen++] = currQuad;
         }
+        _streamReadConstraints.validateNameLength(qlen << 2);
```

Comment:
Yes, this could be instead moved to inside `addName()` which is only called if `findName()` does not find already canonicalized name.